pep-0710.rst

```diff
@@ -0,0 +1,418 @@
+Discussions-To: https://discuss.python.org/t/draft-pep-recording-provenance-of-installed-packages/24838
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 27-Mar-2023
+Post-History: `03-Dec-2021 <https://discuss.python.org/t/pip-installation-reports/12316>`__,
+              `30-Jan-2023 <https://discuss.python.org/t/pre-pep-recording-provenance-of-installed-packages/23340>`__,
+              `14-Mar-2023 <https://discuss.python.org/t/draft-pep-recording-provenance-of-installed-packages/24838>`__,
+
+Abstract
+========
+
+This PEP describes a way to record the provenance of installed Python distributions.
+The record is created by an installer and is available to users in
+the form of a JSON file ``provenance_url.json`` in the ``.dist-info`` directory.
+The mentioned JSON file captures additional metadata to allow recording a URL to a
+:term:`distribution package` together with the installed distribution hash. This
+proposal is built on top of :pep:`610` following
+:ref:`its corresponding canonical PyPA spec <packaging:direct-url>` and
+complements ``direct_url.json`` with ``provenance_url.json`` for when packages
+are identified by a name, and optionally a version.
+
+Motivation
+==========
+
+Installing a Python :term:`Project` involves downloading a :term:`Distribution Package`
+from a :term:`Package Index`
+and extracting its content to an appropriate place. After the installation
+process is done, information about the release artifact used as well as its source
+is generally lost. However, there are use cases for keeping records of
+distributions used for installing packages and their provenance.
+
+Python wheels can be built with different compiler flags or supporting
+different wheel tags.  In both cases, users might get into a situation in which
+multiple wheels might be considered by installers (possibly from different
+package indexes) and immediately finding out which wheel file was actually used
+during the installation might be helpful. This way, developers can use
+information about wheels to debug issues making sure the desired wheel was
+actually installed. Another use case could be tools reporting software
+installed, such as tools reporting a SBOM (Software Bill of Materials), that might
+give more accurate reports. Yet another use case could be reconstruction of the
+Python environment by pinning each installed package to a specific distribution
+artifact consumed from a Python package index.
+
+The motivation described in this PEP is an extension of those in :pep:`610`.  Besides
+stating information about packages installed using a direct URL, installers SHOULD
+record information also for packages installed from Python package indexes when
+identified by their name, and optionally their version.
+
+Rationale
+=========
+
+[describing the reasoning behind the approach chosen in this PEP, and/or a brief history of the discussions that led to this PEP being proposed]
```

Comment:
[Reminder to replace this with an actual Rationale]

Reply:
Done in [2356439](https://github.com/python/peps/pull/3076/commits/2356439ab1dbc8e50d01564664cc8b0c920acc93).